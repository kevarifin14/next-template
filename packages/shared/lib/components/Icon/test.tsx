import { render } from '@testing-library/react';
import React from 'react';

import {
  CheckCircleIcon, ExclamationCircleIcon, XCircleIcon, ExternalLinkIcon, PlusIcon, ArrowsExpandIcon,
} from '.';

describe('icons', () => {
  describe('CheckCircleIcon', () => {
    it('matches snapshot', () => {
      const { baseElement } = render(<CheckCircleIcon />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('ExclamationCircleIcon', () => {
    it('matches snapshot', () => {
      const { baseElement } = render(<ExclamationCircleIcon />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('XCircleIcon', () => {
    it('matches snapshot', () => {
      const { baseElement } = render(<XCircleIcon />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('ExternalLinkIcon', () => {
    it('matches snapshot', () => {
      const { baseElement } = render(<ExternalLinkIcon />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('PlusIcon', () => {
    it('matches snapshot', () => {
      const { baseElement } = render(<PlusIcon />);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('ArrowsExpandIcon', () => {
    it('matches snapshot', () => {
      const { baseElement } = render(<ArrowsExpandIcon />);
      expect(baseElement).toMatchSnapshot();
    });
  });
});
