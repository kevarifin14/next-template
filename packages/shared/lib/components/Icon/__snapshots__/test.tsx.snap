// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icons ArrowsExpandIcon matches snapshot 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="text-gray-500 h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M3 4a1 1 0 011-1h4a1 1 0 010 2H6.414l2.293 2.293a1 1 0 01-1.414 1.414L5 6.414V8a1 1 0 01-2 0V4zm9 1a1 1 0 110-2h4a1 1 0 011 1v4a1 1 0 11-2 0V6.414l-2.293 2.293a1 1 0 11-1.414-1.414L13.586 5H12zm-9 7a1 1 0 112 0v1.586l2.293-2.293a1 1 0 011.414 1.414L6.414 15H8a1 1 0 110 2H4a1 1 0 01-1-1v-4zm13-1a1 1 0 011 1v4a1 1 0 01-1 1h-4a1 1 0 110-2h1.586l-2.293-2.293a1 1 0 011.414-1.414L15 13.586V12a1 1 0 011-1z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`icons CheckCircleIcon matches snapshot 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="text-green-500 h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10 18a8 8 0 100-16 8 8 0 000 16zm3.707-9.293a1 1 0 00-1.414-1.414L9 10.586 7.707 9.293a1 1 0 00-1.414 1.414l2 2a1 1 0 001.414 0l4-4z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`icons ExclamationCircleIcon matches snapshot 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="text-yellow-500 h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M18 10a8 8 0 11-16 0 8 8 0 0116 0zm-7 4a1 1 0 11-2 0 1 1 0 012 0zm-1-9a1 1 0 00-1 1v4a1 1 0 102 0V6a1 1 0 00-1-1z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`icons ExternalLinkIcon matches snapshot 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="text-gray-500 h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11 3a1 1 0 100 2h2.586l-6.293 6.293a1 1 0 101.414 1.414L15 6.414V9a1 1 0 102 0V4a1 1 0 00-1-1h-5z"
      />
      <path
        d="M5 5a2 2 0 00-2 2v8a2 2 0 002 2h8a2 2 0 002-2v-3a1 1 0 10-2 0v3H5V7h3a1 1 0 000-2H5z"
      />
    </svg>
  </div>
</body>
`;

exports[`icons PlusIcon matches snapshot 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10 3a1 1 0 011 1v5h5a1 1 0 110 2h-5v5a1 1 0 11-2 0v-5H4a1 1 0 110-2h5V4a1 1 0 011-1z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;

exports[`icons XCircleIcon matches snapshot 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="text-red-500 h-6 w-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clip-rule="evenodd"
        d="M10 18a8 8 0 100-16 8 8 0 000 16zM8.707 7.293a1 1 0 00-1.414 1.414L8.586 10l-1.293 1.293a1 1 0 101.414 1.414L10 11.414l1.293 1.293a1 1 0 001.414-1.414L11.414 10l1.293-1.293a1 1 0 00-1.414-1.414L10 8.586 8.707 7.293z"
        fill-rule="evenodd"
      />
    </svg>
  </div>
</body>
`;
